core/scripts.py

```diff
@@ -2861,16 +2846,28 @@ def process_task(self, task: DeferredTask):
             for file in files:
                 os.remove(file.name)
 
-    def generate_report(self, data_source_name: str, library_id: int) -> str:
-        """Generate a csv file and return the file path"""
-        with tempfile.NamedTemporaryFile(
-            "w",
-            delete=False,
-        ) as temp:
+    def prepare_email_attachment(self, attachments, file_path, attachment_name, files):
+        """
+        extract contents of files and prepare in a dictionary of email attachments
+        """
+        with open(file_path) as the_file:
+            attachments[attachment_name] = the_file.read()
```

Comment:
This file can potentially be quite large right? I don't think we should read its whole contents into memory here, thats going to blow up the memory usage of this process.